# Burton lab process for extracting DNA for microbiome sequencing
###### Kait Al, PhD. 2023


## Overview

This a modified protocol for the Qiagen DNeasy 96 PowerSoil Pro Kit with notes specific for Burton Lab's equipment.

Sample loading will be sample-type-specific (see step 2). See the following reference for urine and stent -specific info:

Al KF, Burton JP. Processing human urine and ureteral stents for 16S rRNA amplicon sequencing. STAR Protocols 2, 100435. [Link](https://doi.org/10.1016/ j.xpro.2021.100435)



Important notes before starting: 
* Use a fresh lab coat every time
* Wear gloves
* Sterilize the BSC with UV and RNAse Away
* Spray out the centrifuge and rotor with RNAse Away
* Turn on the dry heat bath to ~85 °C (in reality this will be 60 °C)
* If Solution CD3 has precipitated, heat at 60 °C until the precipitate dissolves
* Heat solution C6 to 60 °C before final elution
* Perform all centrifugation steps at room temperature (15-25 °C)
* Use the electronic multichannel pipettor (Unit 1) and DNA extraction-specific Gilson pipettes stored in the extraction room
* Use PCR-grade filter tips for all steps


## Procedure

1. Centrifuge a PowerBead Pro Plate briefly to ensure that the beads have settled at the bottom. (1 minute at max speed, 3700 RPM or 2250 x g in eppendorf centrifuge)

2. Remove and discard the square well mat from the PowerBead Pro Plate. Add up to 0.25 g of soil sample or up to 0.1 g of stool sample in each well. 
	Note: Avoid cross contamination between sample wells. All utensils for sample handling should be 10% bleach and RNAse Away sterilized (forceps, scissors, razor blades, mortar and pestle, etc.)
	1. Stool samples: use a filter pipette tip to scoop up approximately 0.2 g sample and scrape it off the tip into the base of the well.
	2. Stool on toilet paper: use sterilized forceps and scissors to dissect off a single-ply piece of visibly soiled TP approximately 1 cm x 1 cm, and transfer into the well.
	3. Swabs: should be polyurethane foam, not cotton or wood. Use sterilized scissors to cut the swab head directly into the well.
	4. Urine: Should be stored as a frozen pellet. Add 100 μl PCR grade water to resuspend the pellet, and pipette into the well. For very large pellets, 200 μl water may be required.
	5. Kidney stones: Using sterilized forceps, transfer stone to a sterile cell strainer mounted onto an empty 50 ml tube. Rinse by pipetting 1 ml of PCR-grade water over the external surface. Using freshly-sterilized forceps, transfer the stone to a sterilized mortar and pestle and pulverize into sand-like fragments. Suspend fragments in 100 μl PCR-grade water and pipette into the well.
	6. Stents: Use a sterilized razor to slice a 1 cm segment from the proximal and distal ends. Rinse external surface by pipetting 1 ml of PCR-grade water over the external surface. Sterilize the razor again and slice open the rinsed segment length-wise to expose the inner lumen, then transfer into the well.

3. Add 800 μl of Solution CD1 to the wells of the PowerBead Pro Plate. (Option to add less for low abundance samples that aren’t on swabs such as midstream urine pellets).

4. Remove any residual liquid on top of the plate and secure the sealing film (provided) tightly onto the bead plate. Use a tool such as a scraper or plate roller to make the sealing film adhere firmly onto the plate. 
Note: Liquid on top of the plate will prevent a tight sealing of the plate with the sealing film. A strong seal is essential to prevent leakage during disruption in the TissueLyser II.  

5. Put the silicone compression mat (provided) on top of the bead plate that is sealed with the sealing film. 
Note: Two silicone compression mats are provided so that 2 plates can be processed in parallel in the TissueLyser II. The mats are reused for the remaining plates in the kit.

6. Place this entire assembly between 2 Plate Adapter Sets for disruption in the TissueLyser II. 
Important: When using this assembly, do not exceed the recommended disruption time and setting of 2 x 5 min at 25 Hz, because extended processing might lead to leakage. 

7. Shake at a speed of 25 Hz for 5 min. Re-orient the plates so that the side that was closest to the machine body is now furthest from it and shake again at speed of 25 Hz for
5 min. 

8. Centrifuge for 12 minutes.

9. Add 200 μl of Solution CD2 to the collection microtubes. 

10. Discard the sealing film. Turn the pipette aspiration speed down! Otherwise, the tips will get blocked by aspirating beads. TBeing careful to avoid the pellet, transfer the supernatant to the collection microtubes (expect 500–600 μl). The supernatant may still contain soil or stool particles. This step may require going back with a single 1000 μl Gilson pipette to collect supernatant from individual wells. 

11. Seal the collection microtubes with the caps provided and vortex. 

12. Centrifuge for 12 minutes.

13. Add 600 μl of Solution CD3 to each well of an S-block while the sample plate is centrifuging.

14. Transfer up to 700 μl of supernatant to the S-Block (expect 500-600 μl). Avoid disrupting the pellet! If the pellet is disrupted or any is aspirated into the pipette, dispense it back into the original plate and re-centrifuge. NONE of the pellet can be transferred into the next step!

15. Pipet samples up and down to mix. Use caution to avoid overflowing the wells from the volume of the pipette! 

	It is safe to stop the protocol at this step and store the samples covered with 	Sealing Tape (user supplied) at 4 °C.  (Option to incubate plate at 10 mins at 4°C even if not pausing the protocol).
	
16. Place a QIAamp 96 Spin Plate onto a new S-Block. If protocol was paused with plates at 4 °C, centrifuge briefly before removing plate seal.

17. Load approximately 650 μl into each well of the QIAamp 96 Spin Plate and seal the plate with a sealing tape. 

18. Centrifuge for 6 minutes. 

19. Discard the flow-through in the sink with vigorous shaking and place the QIAamp 96 Spin Plate back on the same S-Block. Be very careful not to touch the bottom of the filter to anything (including the edges of the S block). Discard the sealing tape. 

20. Repeat steps 17-19 until all the supernatant has been processed. Discard the final flow-through. 

If you haven't already, turn on the heat block and add C6 to bring to 60 °C.

21. Place the QIAamp 96 Spin Plate back on the same S-Block. Add 500 μl of Solution EA to each well of the QIAamp 96 Spin Plate and seal the plate with a sealing tape.

22. Centrifuge for 6 minutes. Discard the flow-through, and place the QIAamp 96 Spin Plate back on the same S-Block.

23. Add 500 μl of Solution C5 to the QIAamp 96 Spin Plate and seal the plate with a sealing tape. Centrifuge for 6 min. 

24. Discard the flow-through in the sink and remove any trace of the flow-through from the S-Block. Place the QIAamp 96 Spin Plate (still sealed from step 23) into the same S-Block. 

Note: To remove remnants of the flow-through, place a clean paper towel on top of the S-Block or centrifuge the S-Block before placing the QIAamp 96 Spin Plate back into the S-Block.  This step removes residual Solution C5. It is critical to remove all traces of the wash solution because it can interfere with downstream DNA applications. 

25. Centrifuge for 10 minutes. 

26. While the plate is centrifuging, clean out detritus from the BSC and spray down the surface with RNAse Away. 

27. Carefully place the QIAamp 96 Spin Plate onto collection microtubes. Discard the sealing tape. 

28. Allow to air dry inside the hood for 10 min at room temperature.

29. Add 50 μl of ~60 °C Solution C6 to the centre of each well (directly on top of the filter, without touching it). Seal the plate with a sealing tape. Allow to sit for 10 minutes. Depending on downstream requirements you could also elute in water. For high abundance samples, elute in 100 μl instead of 50 μl.

30. Centrifuge for 6 minutes. 

31. Seal collection microtubes with the caps provided using sterile forceps (do not touch the underside of the caps!). OR transfer to an Axygen PCR plate and seal with a foil plate cover. The DNA is now ready for downstream applications. 
Note: Store the DNA frozen (−20 °C or −80 °C), as Solution C6 does not contain EDTA. To concentrate DNA, please refer to the Qiagen Troubleshooting Guide.

32. Label the microtube caps with a sharpie to indicate the column and A/H for directionality. 

33. When using DNA in downstream applications, remove one column's cap at a time (when possible) with sterile forceps onto a sterile surface, and replace in the same directionality. If all caps must be removed at once, sterilize a sheet of aluminum foil inside the BSC, and label it 1-12 with a sharpie. Remove the caps and place them on the foil in their indicated position, then replace them in the correct column and directionality when complete.  

